packages/core/src/request-builder/header-builder/header-builder.ts

```diff
@@ -21,89 +21,80 @@ import { addCsrfTokenAndCookies } from './csrf-token-header';
+    ...csrfHeaders,
+    ...defaultHeaders,
+    ...request.config.customHeaders
+  };
 }
 
 /**
- * Builds the authorization, proxy authorization and SAP client headers for a given destination.
+ * Builds the authorization, proxy authorization and SAP headers for a given destination.
  *
  * @param destination - A destination.
+ * @param customHeaders - Custom default headers for the resulting HTTP headers.
  * @returns HTTP headers for the given destination.
  */
-export const buildHeadersForDestination = (
-  destination: Destination
-): Promise<MapType<string>> =>
-  asyncPipe(
-    addSapClientHeader(destination.sapClient || undefined),
-    // TODO the proxy header are for OnPrem auth and are now handled correctly in the authorization-header.ts and should be removed here
-    // However this would be a breaking change, since we recommended to use 'NoAuthentication' to achieve principal propagation as a workaround.
-    addProxyHeaders(destination.proxyConfiguration),
-    addLocationIdHeader(destination.cloudConnectorLocationId),
-    buildAndAddAuthorizationHeader(destination)
-  )({});
-
-export function addSapClientHeader(sapClient?: string) {
-  return (headers: MapType<string>) =>
-    assocSome('sap-client', sapClient)(headers);
-}
-
-export const addProxyHeaders = (proxyConfiguration?: ProxyConfiguration) => (
-  headers: MapType<string>
-) => mergeSome(headers, path(['headers'], proxyConfiguration));
-
-export const addLocationIdHeader = (locationId?: string) => (
-  headers: MapType<string>
-) => assocSome('SAP-Connectivity-SCC-Location_ID', locationId)(headers);
-
-const buildHeadersAuthFirstCustomLast = <RequestT extends ODataRequestConfig>(
-  request: ODataRequest<RequestT>
-) => (...fns): Promise<MapType<string>> =>
-  asyncPipe(
-    headers => addAuthorizationHeader(request, headers),
-    ...fns,
-    addCustomHeaders(request.config.customHeaders)
-  )({});
-
-const addEtagHeader = (requestConfig: ODataRequestConfig) => (
-  headers: MapType<string>
-) => assocSome('if-match', getETagHeader(requestConfig))(headers);
-
-const addCustomHeaders = (customHeaders: MapType<string>) => (
-  headers: MapType<string>
-) => ({
-  ...headers,
-  ...customHeaders
-});
-
-const addContentTypeHeader = (contentType: string, headers: MapType<string>) =>
-  assoc('Content-Type', contentType)(headers);
+export async function buildHeadersForDestination(
+  destination: Destination,
+  customHeaders?: MapType<any>
+): Promise<MapType<string>> {
+  const authHeaders = await getHeaderByKeyOrExecute(
+    'authorization',
+    customHeaders,
+    () => buildAuthorizationHeader(destination)
+  );
 
-const addAcceptHeader = (headers: MapType<string>) =>
-  assoc('Accept', 'application/json')(headers);
+  const sapHeaders = replaceDuplicateKeys(
+    filterNullishValues({
+      'sap-client': destination.sapClient,
+      'SAP-Connectivity-SCC-Location_ID': destination.cloudConnectorLocationId
+    }),
+    customHeaders
```

Comment:
also here spread?

Reply:
not possible because of lower case and upper case